src/Adapter/Cart/CommandHandler/UpdateProductQuantityInCartHandler.php

```diff
@@ -194,6 +195,16 @@ private function assertProductIsInStock(Product $product, UpdateProductQuantityI
             return;
         }
 
+        if (Pack::isPack($product->id)) {
+            $isEnoughQuantity = Pack::isInStock($product->id, $command->getNewQuantity());
+
+            if (!$isAvailableWhenOutOfStock && !$isEnoughQuantity) {
+                throw new ProductOutOfStockException(sprintf('Product with id "%s" is out of stock, thus cannot be added to cart', $product->id));
```

Comment:
Should we introduce a `PackOutOfStockException` (that would extend `ProductOutOfStockException` ) ? This would allow to tell the two behaviours apart and potentially adapt the error message more accurately